File: app/src/main/java/com/example/wheelsapp/services/APIAuthService.java
package com.example.wheelsapp.services;

import static com.example.wheelsapp.LaunchActivity.TOKEN_KEY;

import com.example.wheelsapp.MainActivity2;
import com.example.wheelsapp.entities.api_request.LoginRequest;
import com.example.wheelsapp.entities.api_responses.LoginResponse;
import com.example.wheelsapp.entities.api_responses.UserResponse;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface APIAuthService {

    @POST("auth")
    Call<LoginResponse> login(@Body LoginRequest login);
}

File: app/src/main/java/com/example/wheelsapp/services/APIAuthService.kt
package com.example.wheelsapp.services

import com.example.wheelsapp.entities.api_request.UserRequest
import com.example.wheelsapp.entities.api_responses.AuthResponse
import retrofit2.Call
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.FormUrlEncoded
import retrofit2.http.POST

interface APIAuthService {
    @POST("/auth")
    @FormUrlEncoded
    fun getToken(@Body user : UserRequest):Response<AuthResponse>
}